identity-did/src/document/core_document.rs

```diff
@@ -56,23 +56,32 @@ pub struct CoreDocument<D = CoreDID, T = Object, U = Object, V = Object>
   #[serde(default = "Default::default", rename = "alsoKnownAs", skip_serializing_if = "OrderedSet::is_empty")]
   pub(crate) also_known_as: OrderedSet<Url>,
   #[serde(default = "Default::default", rename = "verificationMethod", skip_serializing_if = "OrderedSet::is_empty")]
-  pub(crate) verification_method: OrderedSet<VerificationMethod<D,U>>,
+  pub(crate) verification_method: OrderedSet<VerificationMethod<D, U>>,
   #[serde(default = "Default::default", skip_serializing_if = "OrderedSet::is_empty")]
-  pub(crate) authentication: OrderedSet<MethodRef<D,U>>,
+  pub(crate) authentication: OrderedSet<MethodRef<D, U>>,
   #[serde(default = "Default::default", rename = "assertionMethod", skip_serializing_if = "OrderedSet::is_empty")]
-  pub(crate) assertion_method: OrderedSet<MethodRef<D,U>>,
+  pub(crate) assertion_method: OrderedSet<MethodRef<D, U>>,
   #[serde(default = "Default::default", rename = "keyAgreement", skip_serializing_if = "OrderedSet::is_empty")]
-  pub(crate) key_agreement: OrderedSet<MethodRef<D,U>>,
+  pub(crate) key_agreement: OrderedSet<MethodRef<D, U>>,
   #[serde(default = "Default::default", rename = "capabilityDelegation", skip_serializing_if = "OrderedSet::is_empty")]
-  pub(crate) capability_delegation: OrderedSet<MethodRef<D,U>>,
+  pub(crate) capability_delegation: OrderedSet<MethodRef<D, U>>,
   #[serde(default = "Default::default", rename = "capabilityInvocation", skip_serializing_if = "OrderedSet::is_empty")]
-  pub(crate) capability_invocation: OrderedSet<MethodRef<D,U>>,
+  pub(crate) capability_invocation: OrderedSet<MethodRef<D, U>>,
   #[serde(default = "Default::default", skip_serializing_if = "OrderedSet::is_empty")]
-  pub(crate) service: OrderedSet<Service<D,V>>,
+  pub(crate) service: OrderedSet<Service<D, V>>,
   #[serde(flatten)]
   pub(crate) properties: T,
 }
 
+macro_rules! method_mut_ref {
+  ($doc:ident, $method: ident, $query: ident) => {
+    match $doc.$method.query_mut($query.into())? {
+      MethodRef::Embed(method) => Some(method),
+      MethodRef::Refer(ref did) => $doc.verification_method.query_mut(did),
+    }
+  };
+}
```

Comment:
Nit: suggest `method_ref_mut_helper` as the name, also a brief comment why this is needed would be appreciated.

```suggestion
// Workaround for lifetime issues with a mutable reference to self preventing closures from being used.
macro_rules! method_ref_mut_helper{
  ($doc:ident, $method: ident, $query: ident) => {
    match $doc.$method.query_mut($query.into())? {
      MethodRef::Embed(method) => Some(method),
      MethodRef::Refer(ref did) => $doc.verification_method.query_mut(did),
    }
  };
}
```